unified-codes/tools/scripts/dgraph/migration/v2/DataParser.ts

```diff
@@ -1,71 +1,57 @@
 import * as csv from 'csv-parser';
-import * as fs from "fs";
+import * as fs from 'fs';
 
 interface IRow {
-    product: string;
-    //product_synonym,
-    combination: string;
-    route: string;
-    dose_form: string;
-    dose_qualification: string;
-    strength: string;
-    //unit_of_presentation,
-    //immediate_packaging,
-    //pack_size,
-    //outer_packaging,
-    uc1: string;
-    uc2: string;
-    uc3: string;
-    uc4: string;
-    uc5: string;
-    uc6: string;
-    //uc7,
-    //uc8, 
-    //uc9
+  product: string;
+  //product_synonym,
+  combination: string;
+  route: string;
+  dose_form: string;
+  dose_qualification: string;
+  strength: string;
+  //unit_of_presentation,
+  //immediate_packaging,
+  //pack_size,
+  //outer_packaging,
+  uc1: string;
+  uc2: string;
+  uc3: string;
+  uc4: string;
+  uc5: string;
+  uc6: string;
+  uc7: string;
+  //uc8,
+  //uc9,
+  rxnav: string;
 }
 
 type IData = IRow[];
 
+interface IProperty {
+  type: string;
+  value: string;
+}
 interface INode {
-    code: string;
-    name?: string;
-    type?: string;
-    combines?: INode[];
-    properties?: INode[];
-    children?: INode[];
+  code: string;
+  name?: string;
+  type?: string;
+  combines?: INode[];
+  properties?: IProperty[];
```

Comment:
Oops, maybe some confusion here. This is a "generic" type (to allow for the most flexible recursion). I think a more elegant/consistent solution could be to add `value?: string;` to `INode` and to revert properties back to `INode[]`.

Does that make sense?

Reply:
Makes sense. though it is in contrast to the rest of this codebase where we've used specific interfaces rather than using generics. 
Changed and ran a quick test, seems to be ok.  5d807f342

Reply:
Yeah, I think the idea was to be as flexible as possible, since the `csv` format was so fluid. Also I think my idea was that it was good to distinguish the types representing the structure of the input file from the final processed types.